cli/src/formatter.rs

```diff
@@ -1244,4 +1261,41 @@ mod tests {
             String::from_utf8(output).unwrap(),
             @"<< outer1 >>[38;5;1m<< inner1  inner2 >>[39m<< outer2 >>");
     }
+
+    #[test]
+    fn test_raw_format_recorder() {
+        // Note: similar to test_format_recorder above
+        let mut recorder = FormatRecorder::new();
+        write!(recorder.raw(), " outer1 ").unwrap();
+        recorder.push_label("inner").unwrap();
+        write!(recorder.raw(), " inner1 ").unwrap();
+        write!(recorder.raw(), " inner2 ").unwrap();
+        recorder.pop_label().unwrap();
+        write!(recorder.raw(), " outer2 ").unwrap();
+
+        // No non-raw output to label
+        let config = config_from_string(r#" colors.inner = "red" "#);
+        let mut output: Vec<u8> = vec![];
+        let mut formatter = ColorFormatter::for_config(&mut output, &config, false).unwrap();
+        recorder.replay(&mut formatter).unwrap();
+        drop(formatter);
+        insta::assert_snapshot!(
+            String::from_utf8(output).unwrap(), @" outer1  inner1  inner2  outer2 ");
```

Comment:
I think this is actually a bug. If `.push_label()` were eagerly evaluated (or we had non-raw text there), `inner1  inner2` would be labeled.

For example, `label("A", raw_escape_sequence("B") ++ "content" ++ raw_escape_sequence("/B"))` should be rendered as `<A><B>content</B></A>`, not `<B><A>content</B></A>`.

Can you leave a TODO comment?